agreementbot/agreementbot.go

```diff
@@ -1542,9 +1549,116 @@ func (w *AgreementBotWorker) secretsUpdate() int {
 	return 0
 }
 
+func (w *AgreementBotWorker) handleHAGroupChange(msg *events.ExchangeChangeMessage) error {
+	glog.Info(AWlogString(fmt.Sprintf("AgreementBot start to handle HA group change: %v", msg.String())))
+	change := msg.GetChange()
+	deletedCache := msg.GetResourceBeforeChange()
+
+	glog.Info(AWlogString(fmt.Sprintf("AgreementBot is handling HA group change: %v, deleted cache is: %v", change, deletedCache)))
+	haGroupChange, _ := change.(exchange.ExchangeChange)
+
+	var deletedHACache *exchangecommon.HAGroup
+	if castDeletedHACache, ok := deletedCache.(exchangecommon.HAGroup); ok {
+		deletedHACache = &castDeletedHACache
+	} else {
+		deletedHACache = nil
+	}
+
+	if haGroupChange.Operation == exchange.CHANGE_OPERATION_CREATED {
+		// save cache
+		if exHAGroup, err := GetHAGroup(haGroupChange.OrgID, haGroupChange.ID, w.GetHTTPFactory().NewHTTPClient(nil), w.GetExchangeURL(), w.GetExchangeId(), w.GetExchangeToken()); err != nil {
```

Comment:
No need to save the cache here because next GetHAGroup function will get the latest and save to cache.